bundles/http/Citrix/7150aa4a-3bcd-4a73-a15c-893249515184/101b58cb-1023-48a1-9aca-4b64c2dbfd68.js

```diff
@@ -0,0 +1,402 @@
+const pageSize = 100
+
+function fullSync ({ client, dataStore, integrationParameters }) {
+  const credentials = getCredentials(client, integrationParameters)
+
+  syncProjects({client, dataStore, credentials})
+  syncUsers({client, dataStore, credentials})
+  syncViews({ client, dataStore, credentials })
+  syncWorkbooks({client, dataStore, credentials})
+}
+
+async function syncProjects({ client, dataStore, credentials }) {
+  let pageNumber = 0
+  let totalAvailable = 0
+
+  do {
+    pageNumber++
+
+    console.log('projects page ' + pageNumber)
+
+    await request(client, 'projects', credentials, pageNumber)
+    .then(body => {
+      totalAvailable = parseInt(body.pagination.totalAvailable)
+      const projects = body.projects.project
+      
+      if (body.projects !== undefined && Array.isArray(projects) && projects.length > 0){
+        dataStore.save('projects', projects)
+      } else {
+        console.log('WARNING: Endpoint projects - Bad or empty response')
+      }
+      
+      if (body.pagination == undefined || totalAvailable == null||undefined) {
```

Comment:
Please don't `totalAvailable == null||undefined`
Check [operator precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence). 
Imho it won't work as you expected. I think that you wanted to do this 
```js
if (body.pagination == undefined || totalAvailable == (null || undefined)) {
```
but this is not very readable. Better is to use 
```js
if (body.pagination == undefined || totalAvailable == null || totalAvailable == undefined)) {
```
But check what is difference between `== null` and `=== null`.

imho this will work as you wanted
```js
if (body.pagination == undefined || totalAvailable == null) {
```
because `(null == undefined) === true` :)

Personally I would write 
```js
if (totalAvailable == null) {
```
I removed `body.pagination == undefined` because on line 23 you already expecting that it exists...
So maybe you should fix it somehow...